ml/jml/glz_classifier.cc

```diff
@@ -363,12 +363,25 @@ explain(const Feature_Set & feature_set,
 
     ExcAssertEqual(decoded.size(), features.size());
 
+    // if we're in regression mode
+    int label_idx = -1;
+    if(weights.size() == 1) {
+        label_idx = 0;
+    }
+    else {
+        label_idx = label;
+    }
+    if(label_idx >= weights.size()) {
+        throw ML::Exception("label bigger than weight vector");
+    }
+
     for (unsigned j = 0;  j < features.size();  ++j) {
         result.feature_weights[features[j].feature]
-            += weight * weights[label][j] * decoded[j];
+            += weight * weights[label_idx][j] * decoded[j];
```

Comment:
Use the Label class; it should be clearer that it's a float for regression and an int for classification.